### ISSUE TITLE
June 2019

### ISSUE THREAD

**Issue by chicoxyzzy** [babel/proposals#58]
https://github.com/tc39/agendas/blob/master/2019/06.md

**Comment by chicoxyzzy** [comment]
Dynamic import moves to stage 4 🎉

**Comment by jridgewell** [comment]
Let all early errors be SyntaxErrors gained consensus

**Comment by jridgewell** [comment]
For-in enumeration order reached Stage 2

**Comment by jridgewell** [comment]
BigInt to Stage 4

**Comment by jridgewell** [comment]
eval(nonString) should not side effect gained consensus

https://github.com/tc39/ecma262/pull/1504

**Comment by jridgewell** [comment]
Set "name" property for anonymous functions gained consensus

**Comment by chicoxyzzy** [comment]
Intl.DisplayNames to stage 2

**Comment by chicoxyzzy** [comment]
Emitter for Stage 1

**Comment by jridgewell** [comment]
Array.isTemplateObject to Stage 2

https://github.com/mikesamuel/proposal-array-is-template-object

**Comment by chicoxyzzy** [comment]
There was also some normative changes to ECMA-402 which got consensus

**Comment by chicoxyzzy** [comment]
Optional chaining to stage 2

Nullish Coalescing to stage 2

**Comment by jridgewell** [comment]
Top level await to Stage 3

**Comment by jridgewell** [comment]
Lookup constructor.resolve only once in PerformPromise{All, Race}  got consensus

**Comment by jridgewell** [comment]
WeakRefs to Stage 3, pending Kevin's editor review

**Comment by zloirock** [comment]
What about the standard library and `Promise.any`?

**Comment by ljharb** [comment]
They’re not mentioned here because they have not changed stage.

**Comment by zloirock** [comment]
Anyway, adding this information could be useful.

**Comment by ljharb** [comment]
That seems silly, a great many things haven’t changed stage.

**Comment by zloirock** [comment]
That seems silly don't write about it since those proposals claimed to change the state at this meeting.

**Comment by ljharb** [comment]
They sought advancement on the agenda; but not everything that seeks it gets it.

The purpose of this issue isn’t to update people on the meeting - that’s what the notes are for. It’s to let Babel people know what actions they might need to take as a result of the meeting.

**Comment by zloirock** [comment]
...or not needed.

**Comment by ljharb** [comment]
None are ever needed, unless otherwise indicated.

**Comment by zloirock** [comment]
@ljharb I can't understand, is it too hard to write the summary of the result of discussion for each proposal from the agenda? I'm one of the first implementors of proposals - and since I haven't any direct information from TC39 meetings, I should keep in mind that the state of the proposal was able to be changed and just someone forgot to publish it here or somewhere else before publishing in the meeting notes repo.

**Comment by ljharb** [comment]
Yes, it is - the agenda has many things on it, and the *notes* - which you just have to wait for - are that writeup.

The [tc39/proposals](https://github.com/tc39/proposals) repo is the closest thing to a source of truth; that's where it will be up to date after every meeting.

**Comment by zloirock** [comment]
...which also contains obsolete information and updated with a delay.

Awesome, thanks.

**Comment by ljharb** [comment]
What information in it is obsolete? and yes, a delay is fine - there's no rush on adapting to these changes.

**Comment by zloirock** [comment]
I wrote about issues or messages to required people about obsolete information in `tc39/proposals` repo enough. But now I write about this repo as about a place where I can get immediate information about changes in states of proposals for planning releases / changes in `core-js`. I can't understand, it's too hard to write just some words that after discussion that the state of the proposal was not updated? It could be very useful to me.

**Comment by ljharb** [comment]
If you don't see a message saying it was updated, then it wasn't updated. I'm not sure what's so hard about that.

The proposals repo is fully up to date. If you have an issue to file, please file it, but otherwise whatever's in that repo is the state of all proposals.

**Comment by zloirock** [comment]
>  If you don't see a message saying it was updated, then it wasn't updated.

See the previous issues - not all updates were published here. Otherwise, such questions would not arise.

> The proposals repo is fully up to date.

Also not always.

**Comment by ljharb** [comment]
I'm the one tasked to keep it so; and yes, it is up to date, as far as I'm aware.

**Comment by zloirock** [comment]
Seems you did not hear me. I still don't see any updates even in dynamic import or `BigInt` proposal, I remember when some proposal repos were not updated more than half a year. I don't see any updates in `tc39/proposals` - as you wrote, there's no rush on adapting to these changes. But not in my cases, where immediate information is very useful.

**Comment by ljharb** [comment]
That's because those proposals are not, technically, at stage 4 until their spec PRs were approved, per the process document. I pushed https://github.com/tc39/proposals/commit/b04295ff4af51c143c62df62947ae7cb25759691 the instant I approved `import()` - which was the first moment it was actually stage 4, so you can't get any more up to date than that.. Any announcement otherwise was premature. BigInt is still at stage 3, because its PR is not yet approved.
